plugins/ltac/pptactic.ml

```diff
@@ -1349,16 +1336,6 @@ let () =
     (lift_env (fun env sigma (c, _) -> pr_glob_constr_pptac env sigma c))
     (make_constr_printer Printer.pr_econstr_n_env)
   ;
-  Genprint.register_print0
```

Comment:
On second thought, I think this is a regression to transfer this registering from the ltac plugin to vernac. The red_expr generic argument is only used in the former, so instead of moving this code down, you should move the red_expr genarg up. If possible, it should probably belong to Extraargs.

Reply:
And where do you declare the printer for vernac-level reduction expressions? (Assuming we are not redoing an old discussion.)

Reply:
Leave it where it is right now?

Reply:
So, no printability of `Definition foo := Eval compute in bar`?

Reply:
What has this to do with the current change? The command is not defined in terms of the generic argument at hand.

Reply:
Ping.

Reply:
Sorry, didn't we conclude at some time that `Definition foo := Eval compute in bar.` was part of the specification language, independently of having required Ltac or not?

Reply:
I think that it was rather the contrary. The genarg implicitly refers to Ltac semantics for the arguments. Furthermore, your code shows that it's currently only used in the Ltac plugin, and there would be no point in lowering as a vernac argument since the `Eval` definitional command uses a dedicated mechanism not relying on generic arguments (and it can't because it needs special parsing).

Reply:
You're right, `Eval` does not rely on `genarg`. But then, why to have the `register_grammar Genredexpr.wit_red_expr` in `pvernac.ml` (and same for `wit_binders`)? If the only consumer of `wit_redexpr` and `wit_binders` is Ltac1, not only the printer but the generic grammar could also go in `plugins/ltac`?

Reply:
Yes, this is actually my point.

Reply:
Then I move also the `register_grammar` which was originally in `pvernac` to ltac1, agreed?

Reply:
Indeed.

Reply:
Done.